lib/block_view_lockups.go

```diff
@@ -1303,6 +1303,36 @@ func (bav *UtxoView) _connectCoinLockup(
 			}
 		}
 
+		// (1.5) Verify  transfer restriction statuses as being respected
+
+		// Fetch the transfer restrictions attached to the transfer.
+		transferRestrictionStatus := profileEntry.DAOCoinEntry.LockupTransferRestrictionStatus
+
+		// Fetch the "sender" (transactor's) PKID entry.
+		senderPKIDEntry := bav.GetPKIDForPublicKey(txn.PublicKey)
+		if senderPKIDEntry == nil || senderPKIDEntry.isDeleted {
+			return 0, 0, nil,
+				errors.Wrap(RuleErrorCoinLockupInvalidSenderPKID, "_connectCoinLockup")
+		}
+		senderPKID := senderPKIDEntry.PKID
+
+		// Check if transfer restrictions are respected from the lockup.
+		if transferRestrictionStatus == TransferRestrictionStatusProfileOwnerOnly && !profilePKID.Eq(senderPKID) {
+			return 0, 0, nil,
+				errors.Wrap(RuleErrorCoinLockupRestrictedToProfileOwner, "_connectCoinLockup")
+		}
+
+		// Check if the lockup respects transfer limited to DAO members only.
+		// Here, a "DAO member" is anyone who holds either unlocked or locked DAO coins associated with the profile.
+		if transferRestrictionStatus == TransferRestrictionStatusDAOMembersOnly {
+			receiverBalanceEntry :=
+				bav._getBalanceEntryForHODLerPKIDAndCreatorPKID(hodlerPKID, profilePKID, true)
+			if receiverBalanceEntry.BalanceNanos.IsZero() && lockedBalanceEntry.BalanceBaseUnits.IsZero() {
```

Comment:
I think you need to check nil or isDeleted here.

Reply:
In DBGetBalanceEntryForHODLerAndCreatorPKIDsWithTxn which is called from the _getBalanceEntryForHODLerPKIDAndCreatorPKID function, we return an empty balance entry with zero nanos as the balance. I believe this is the only place in the code where we do this pattern, but it's not necessary to check if it's nil here as a reuslt.